lib/src/main/java/com/diffplug/spotless/java/ImportOrderStep.java

```diff
@@ -106,15 +117,23 @@ private static final class State implements Serializable {
 		private final List<String> importOrder;
 		private final String lineFormat;
 		private final boolean wildcardsLast;
+		private final boolean semanticSort;
+		private final Set<String> treatAsPackage;
+		private final Set<String> treatAsClass;
```

Comment:
These should be `TreeSet<String>`. We do equality check using serialized representation, so if HashSets end up with different item ordering then up-to-date stuff won't work (e.g. linux CI runners populating buildcache vs a Windows dev machine).